File: C++/First Steps in Coding - Basics of C++/While Loops/Exercise/Vacation.cpp
#include <iostream>
#include <string>
#include <iomanip>
using namespace std;

int main() {
	int daysSpending = 0, totalDays=0;
	double requiredMoney, availableMoney;
	cin >> requiredMoney; cin >> availableMoney;
	while (true) {
		string action;
		cin >> action;
		if (action == "spend") {
			double sum; cin >> sum;
			if (sum >= availableMoney) {
				availableMoney = 0;
			}
			else {
				availableMoney -= sum;
			}
			daysSpending += 1;
		}
		else if (action == "save") {
			daysSpending = 0;
			double sum; cin >> sum;
			availableMoney += sum;
		}
		totalDays += 1;
		if (daysSpending == 5) {
			cout << "You can't save the money.\n" << totalDays;
			break;
		}
		if (availableMoney >= requiredMoney) {
			cout << "You saved the money for " << totalDays << " days.";
			break;
		}
	}
}

File: Python/Basics/While Loop/Exercise/Vacation.py
vacationMoney = float(input())
availableMoney = float(input())
consDays = 0
days = 0

while True:
    #spend/save
    action = input()
    amount = float(input())

    if action == "spend":
        if amount > availableMoney:
            availableMoney = 0
        else:
            availableMoney -= amount
        consDays += 1

    elif action == "save":
        availableMoney += amount
        consDays = 0


    days += 1

    if consDays == 5:
        print(f"You can't save the money.\n{days}")
        break

    if availableMoney >= vacationMoney:
        print(f"You saved the money for {days} days.")
        break